presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java

```diff
@@ -512,64 +504,71 @@ void reset()
             }
         }
 
-        private boolean test(long value)
+        int adjustValue(long value, int offsetIndex)
+        {
+            if (inDictionaryStream != null) {
+                if (!inDictionaryFlags[offsetIndex]) {
+                    value += stripeDictionarySize;
```

Comment:
let's not modify input parameters and let's combine two if statements for readability:

```
            if (inDictionaryStream != null && !inDictionaryFlags[offsetIndex]) {
                return toIntExact(value) + stripeDictionarySize;
            }
            
            return toIntExact(value);
```